core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -361,6 +383,55 @@ public void close() throws IOException {
     httpClient.close(CloseMode.GRACEFUL);
   }
 
+  public static HTTPClient buildFrom(Map<String, String> properties) {
```

Comment:
I know this is intended to clean up, but it now creates two different ways to build a client. I'd prefer to have just one, but I don't really care whether that is to deprecate the builder and only use this method or whether it is to keep the builder and add properties to it.

I have a slight preference for adding properties to the builder since the `S3V4RestSignerClient` uses the `uri` method. Basically, that would mean adding a new `builder(Map<String, String>)` factory method. Then that method would do everything that this method does, but return the builder at the end for overrides.

Reply:
I've added a deprecation to the static builder method so that it's clear which is supported going forward.

I don't feel the builder pattern here is a good approach because the HTTPClient in Iceberg is already tightly coupled to the RESTCatalog implementations due to request/response/exception types and just creates a separate builder on top of the apache httpclient implementation.  We should just construct from the properties map, which will be much easier to extend with new properties going foward.

Reply:
Sounds good to me.

Now that the builder is deprecated, we should remove the other use of it in `S3V4RestSignerClient`.

Reply:
I went back and changed the builder to accept properties because that was easier than updating how the S3V4 signer injects the object map.  I don't love that we've got multiple levels of builder pattern going on, but now that we pass in the properties, we can extract values out of that as needed going forward.